### ISSUE TITLE
Recent dependency bumps causes MsalUiRequiredException, breaks functionality

### ISSUE THREAD

**Issue by lwestfall** [microsoftgraph/msgraph-sample-aspnet-core#129]
**Describe the bug**
The version bumps in the following packages seem to have breaking changes unaccounted for in this repo (I'm not familiar enough to understand what, exactly):
- Microsoft.Identity.Web
- Microsoft.Identity.Web.MicrosoftGraph
- Microsoft.Identity.Web.Ui

Maybe it's something with our Azure App but I consistently get the following exceptions when signing in, and the site is broken after that:

```
MsalUiRequiredException:
        ErrorCode: user_null
Microsoft.Identity.Client.MsalUiRequiredException: No account or login hint was passed to the AcquireTokenSilent call.
```

**To Reproduce**
1. Make sure you've signed out of the application or use private browser
2. Checkout 7aeeb9a26c419ab57b073d88119fa226400406c1 (HEAD at main branch at time of writing)
3. Try to sign in
4. Site is broken:
   - Dropdown menu in upper right is incomplete
   - Welcome message for user on home page missing
   - `TimeZoneNotSupportedException` displayed in calendar (screenshot below)

**Expected behavior**
Checking out 9b6b9c99c32b7f7a486dd03cb8a2c90e93c690c8 (prior to the version bumps above) seems to solve the issue. I can sign in and view my calendar events. No issues

**Screenshots**
![image](https://github.com/microsoftgraph/msgraph-sample-aspnet-core/assets/15251311/a68affbe-7c1a-4132-a509-56ffbc1a9413)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Edge
 - Version: 117.0.2045.60

**Comment by jasonjoh** [comment]
Thanks for reporting this @lwestfall! I reproduce this as well. It seems with the release of 2.15.1 for the Microsoft.Identity.Web dependencies, they [introduced a bug](https://github.com/AzureAD/microsoft-identity-web/issues/2513) that impacts custom `OnTokenValidated` handlers, like the one we use in this sample to add Graph information to the claims principal. I believe this is causing the failure.

For now, I would keep all three Microsoft.Identity.Web dependencies (Microsoft.Identity.Web, Microsoft.Identity.Web.UI, and Microsoft.Identity.Web.MicrosoftGraph) at 2.14.0 until a fix is released.

**Comment by lwestfall** [comment]
@jasonjoh thanks for the quick response and for validating my sanity!

**Comment by crazycga** [comment]
Man, do I feel dumb.  I'll close my ticket and show myself out.  Thanks @jasonjoh .

**Comment by jasonjoh** [comment]
@lwestfall @crazycga Looks like the issue has been resolved in the 2.15.3 versions of the Microsoft.Identity.Web* libraries.
